src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -61,6 +61,14 @@ public Book findOne(@PathVariable final String isbn) {
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
     public Book create(@RequestBody final Book book) {
+        if (book.getIsbn().isEmpty() || book.getTitle().isEmpty() || book.getSubtitle().isEmpty() ||
+            book.getAuthor().isEmpty() || book.getImage().isEmpty() || book.getPublisher().isEmpty() ||
+            book.getYear().isEmpty() || book.getPages() <= 0) {
+            throw new EmptyNeccessaryAttributesException();
+        }
+        if (bookRepository.findByIsbn(book.getIsbn()).isPresent()) {
+            throw new BookAlreadyExistsException(book.getIsbn());
+        }
         return bookRepository.save(book);
```

Comment:
You can do this with a try catch
```suggestion
 try {
       if (bookRepository.findByIsbn(book.getIsbn()).isPresent()) {
            throw new BookAlreadyExistsException(book.getIsbn());
        }
        return bookRepository.save(book);
    } catch (Exception e) {
        throw new failedToCreateBookException(e);
    }
```

If some of the fields that are null will throw NullPointerException and the catch will take that exception.

Reply:
I think you will do a pre-condition in a future card to setters.
For example:
`Preconditions.checkNotNull(genre, "genre cannot be null");`